### DISCUSSION TITLE
Optimizing Avalonia-based Desktop App Size

### DISCUSSION THREAD

**Discussion by Pratham95VP** [AvaloniaUI/Avalonia#14633] [General]
I've developed a Windows desktop application using Avalonia UI 11.0.9. Even with just the blank application and Avalonia.ReactiveUI packages, without including Avalonia.Fonts.Inter or any third-party packages, the size of the executable is approximately 80MB. Can anyone confirm if they're experiencing similar sizes? If so, how can we go about reducing this size?

**Comment by thevortexcloud** [comment]
You can try enabling trimming. But it may break things.

https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/trimming-options

**Reply by Pratham95VP** [reply]
I’ve experimented with it, but unfortunately, it proved ineffective and caused frequent app crashes.

**Reply by timunie** [reply]
Trimming is your only option to reduce app size. Use as less reflection as possible, enable compiled bindings. Or accept the App size.

**Reply by Pratham95VP** [reply]
ok

**Reply by thevortexcloud** [reply]
The next best thing you can do if you can't use trimming is to reduce your external dependencies and rely on things included in the runtime. For example, ReactiveUI is mostly just a wrapper around `INotifyPropertyChanged`. 

But for a desktop app I would not worry too much about only ~80mb. Most people these days wont even notice it. That said, I also just checked the size of one of my real apps. Which has numerous third party depedencies (including Avalonia) and the entire thing is apparently only ~26mb when using a normal platform specific release publish. So something odd does seem to be happening with your build. That or windows dependencies are significantly heavier than non Windows ones. Have you checked the size when using `dotnet publish -r win-x64`?

**Reply by Pratham95VP** [reply]
The size that I have mentioned is without third party dependencies and used ``dotnet publish -r win-x64 -p:PublishSingleFile=true --self-contained true`` this to publish.

**Reply by thevortexcloud** [reply]
Ah, you are using self contained. I was looking at framework dependent builds since they don't include the entire .NET runtime, which is a fairly significant space saver.

**Reply by Pratham95VP** [reply]
Thanks for pointing out. I will set ``--self-contained false`` .
